javascript/src/main/components/Courses/CourseList.js

```diff
@@ -0,0 +1,45 @@
+  
+import React from "react";
+import { Link } from "react-router-dom";
+import BootstrapTable from 'react-bootstrap-table-next';
+import { useHistory } from "react-router-dom";
+import useSWR from "swr";
+import { fetchWithoutToken } from "main/utils/fetch";
+
+export default ({courses}) => {
+    const history = useHistory();
+
+    const { data: filter } = useSWR(
+        "/api/public/filter",
+        fetchWithoutToken
+    );
+
+    if (filter && courses && filter.length > 0 &&filter[0].activeQuarter !== "All") {
+        if (filter.length > 0 &&filter[0].activeQuarter !== "All") {
```

Comment:
These two nested if statements appear to be checking for the same thing. Can we eliminate one of them?

Reply:
I copied these two lines of code from the work of another team to apply their features to ours.
I am not sure why they wrote code like this
But I would prefer to keep it though

Reply:
Is there a reason we need both if statements? I think they're redundant

Reply:
You marked this as resolved again, with no actual resolution. If you don’t plan on fixing this, can you provide your reasoning?

Reply:
The problem you mentioned has been resolved. Ready for review.